baseapp/baseapp.go

```diff
@@ -655,6 +682,12 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 	result = app.runMsgs(ctx, msgs, mode)
 	result.GasWanted = gasWanted
 
+	// consume block gas
+	if mode == runTxModeDeliver {
```

Comment:
OK what happens if the transaction runs out of gas during execution? I think we'll never hit this block... but rather we want to deduct the gas the transaction has actually consumed, so I think we need another `BlockGasMeter().ConsumeGas()` call in the recover function.

Reply:
This makes sense. @rigelrozanski do you have any thoughts here? I think we can extract this into an auxiliary function.